modules/kafka/kafka-dest-driver.c

```diff
@@ -324,6 +412,32 @@ _purge_remaining_messages(KafkaDestDriver *self)
 
 }
 
+static gboolean
+_init_template_topic_name(KafkaDestDriver *self)
+{
+  self->topicname_is_a_template = TRUE;
```

Comment:
I think this variable is not needed at all, and its logic could be replaced with something like:

```
gboolean
kafka_dd_is_topic_name_a_template(KafkaDestDriver *self)
{
  return self->topic == NULL;
}
```